fix to bst_process incorrect resampling error message

Moo-Marc opened the pull request:
as per forum report https://neuroimage.usc.edu/forums/t/19542 and https://github.com/brainstorm-tools/brainstorm3/issues/299 

Optional fix since the revert Francois did also fixed the issue.  

This simply adds an error message if a process attempts to split by columns and change number of samples.  I also moved a few lines that only need to run once.

ftadel commented:
As I explained before, optional fixes to sensitive code are not particularly welcome. 

This doesn't seem very harmful, but at the same time it seems quite useless in the sense it does not bring any improvement to the way the function bst_process.m works. It captures an error that should not occur in the first place and will not bring any speed improvement. On the other hand, it alters a function that has been stable for years in a way that could lead to weird unwanted side effects (a statement that was executed multiple times is not anymore), which can cause sneaky bugs that would take hours to find. 

The previous attempt you mention here took me a while to handle, for zero improvement. 
As we are already have an infinite todo list of useful things to do (https://neuroimage.usc.edu/brainstorm/Next), could we agree on avoiding spending time on useless-but-hazardous modifications in the future?  

If you are 100% sure this is safe and you want to move on with this modification, can you please handle the merge with @martcous?

Thanks

martcous commented:
I agree with François here, it's better to fix the processes themselves rather than write an endless amount of potential error catches in bst_process. A bug in a single process is much less limiting than in a function called by every process. The issue seems to have been fixed by François so I would not move forward with this change.

Moo-Marc commented:
Not an unexpected reaction, but you did ask me, correctly, to triple check.  So I found the error (which was due to how I split my previous pull request) and fixed it. Now up to you which fix you prefer.

ftadel commented:
I'd prefer avoiding additional modifications to this function.
Are you confident that this partial revert I made as an emergency fix does fix all the problems that were introduced in bst_process?

Moo-Marc commented:
I'm aware of only this one problem.  I quickly rechecked issues and the forum.  Did I miss something?

ftadel commented:
I meant: 
Are you confident that this partial revert I made as an emergency fix does entirely fix this problem that was introduced in bst_process?

If you think this OK the way it is now on the repo, I would close this PR.

Moo-Marc commented:
Yes, this problem is fixed by the revert.

I disagree that this error message is useless though.  This is an unchecked and undocumented (without digging into the code) assumption - that resampling is not allowed when splitting by column blocks.

ftadel commented:
OK, it's not useless from the coding point of view.
But in practice very few people write processes, while thousands of people use the software, and we have tons of things very useful from the user point of view to do first :)
